src/components/application_manager/src/commands/command_request_impl.cc

```diff
@@ -986,6 +985,36 @@ void CommandRequestImpl::AddTimeOutComponentInfoToMessage(
   }
 }
 
+void CommandRequestImpl::AddRequestToTimeoutHandler(
+    smart_objects::SmartObject& request) const {
+  uint32_t function_id =
+      request[strings::params][strings::function_id].asUInt();
+  /*SDL must not apply "default timeout for RPCs processing" for
+  BasicCommunication.DialNumber RPC (that is, SDL must always wait for HMI
+  response to BC.DialNumber as long as it takes and not return GENERIC_ERROR
+  to mobile app), so the OnResetTimeout logic is not applicable for DialNumber
+  RPC*/
+  if (hmi_apis::FunctionID::BasicCommunication_DialNumber ==
+      static_cast<hmi_apis::FunctionID::eType>(function_id)) {
+    return;
+  }
+  /*If soft buttons are present in Alert RPC, SDL will not use timeout tracking
+  for response, so the OnResetTimeout logic is not applicable in this case*/
+  if (hmi_apis::FunctionID::UI_Alert ==
+      static_cast<hmi_apis::FunctionID::eType>(function_id)) {
+    if (request.keyExists(strings::msg_params)) {
+      if (request[strings::msg_params].keyExists(strings::soft_buttons)) {
+        return;
```

Comment:
@LitvinenkoIra I think this case should be logged

Reply:
@LuxoftAKutsan logging added 278050fc416a577487a1bd2acbf1eedfdbea3c37